unified-runtime/source/adapters/opencl/event.hpp

```diff
@@ -56,6 +56,7 @@ struct ur_event_handle_t_ {
                                           nullptr));
       UR_RETURN_ON_FAILURE(ur_queue_handle_t_::makeWithNative(
           native_queue, Context, nullptr, Queue));
+      urQueueRetain(Queue);
```

Comment:
not sure this is necessary, `makeWithNative` creates a new handle so we already own the only reference to `Queue`

Reply:
This will mean a mismatched reference count and thus a leak, would it not?

Reply:
Actually, yeah. The constructor of ur_queue_handle_t sets the initial RefCount to 1 so implicitly retains it.